lib/command.js

```diff
@@ -232,23 +239,16 @@ module.exports = function command (yargs, usage, validation, globalMiddleware) {
       const populateDoubleDash = !!yargs.getOptions().configuration['populate--']
       if (!populateDoubleDash) yargs._copyDoubleDash(innerArgv)
 
-      innerArgv = applyMiddleware(innerArgv, yargs, middlewares, false)
-      let handlerResult
-      if (isPromise(innerArgv)) {
-        handlerResult = innerArgv.then(argv => commandHandler.handler(argv))
-      } else {
-        handlerResult = commandHandler.handler(innerArgv)
-      }
-
-      if (isPromise(handlerResult)) {
-        yargs.getUsageInstance().cacheHelpMessage()
-        handlerResult.catch(error => {
-          try {
-            yargs.getUsageInstance().fail(null, error)
-          } catch (err) {
-            // fail's throwing would cause an unhandled rejection.
-          }
-        })
+      try {
+        await applyMiddleware(innerArgv, yargs, middlewares, false)
+        if (commandHandler.handler.length < 2) {
+          await commandHandler.handler(innerArgv)
```

Comment:
above in this code `innerArgv` was overridden but now you don't override value

Reply:
This was a redundant overriding, as applyMiddleware already modifies its first parameter.

So I removed it for it to be more explicit.